pkg/stanza/fileconsumer/internal/archive/archive.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
+
+import (
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+)
+
+const _ = "knownFiles"
+
+type Archive interface {
+	SetStorageClient(operator.Persister)
+	Match([]*FileRecord) ([]*reader.Reader, error)
```

Comment:
Why can this return multiple matches but all our other matching logic only returns one?

Reply:
Essentially, we do the following.

```shell
readers = []
for key in archive {
   records := persister.Get(key)
   for file in unmtached_files {
        if match found in records
             // create a new reader from known metadata
             // add the reader to readers
             // remove the file from unmtached_files
             // also, remove the offset from storage (as a match is found)
   }
  // rewrite the storage back to disk with remaining metadata
}

// unmtached_files would now contain new files 
for file in unmatched_files {
   // create a new reader
   // add the reader to readers
}
return readers
}

```

Just to note, we can achieve the same thing by keeping the reader creation under `newReader`. I think that should be preferred. Let me know what do you think?

Reply:
> Just to note, we can achieve the same thing by keeping the reader creation under newReader. I think that should be preferred. Let me know what do you think?

Yes, let's explore that option further before worrying about this.

Reply:
@djaglowski Makes sense. Let's continue our discussion on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35098